lib/Sema/CSStep.cpp

```diff
@@ -883,11 +883,15 @@ bool ConjunctionStep::attempt(const ConjunctionElement &element) {
   assert(!ModifiedOptions.has_value() &&
          "Previously modified options should have been restored in resume");
   if (CS.isForCodeCompletion() &&
-      !element.mightContainCodeCompletionToken(CS)) {
+      !element.mightContainCodeCompletionToken(CS) &&
+      !element.getLocator()->isForSingleValueStmtConjunctionOrBrace()) {
     ModifiedOptions.emplace(CS.Options);
     // If we know that this conjunction element doesn't contain the code
     // completion token, type check it in normal mode without any special
     // behavior that is intended for the code completion token.
+    // Avoid doing this for SingleValueStmtExprs, because we can more eagerly
+    // prune branches in that case, which requires us to detect the code
+    // completion option while solving the conjunction.
     CS.Options -= ConstraintSystemFlags::ForCodeCompletion;
```

Comment:
@ahoppen Out of curiosity, is this even still needed now that we have 6cec68e302477f30264387a1b0533214aa5d6f8e? Seems like removing this could allow us to skip conjunction elements in more cases.

Reply:
Oh, I think we might be able to remove it. Would you mind opening a PR removing it and running the SourceKit stress tester? I’ll check the stress tester performance with the change.

Reply:
Sure thing: #67626